### ISSUE TITLE
should manifold.time/every and manifold.stream/periodically keep a process running?

### ISSUE THREAD

**Issue by derwolfe** [ztellman/manifold#61]
I am relatively new to using manifold, but am accustomed to writing Twisted. My expectation of `every` and `periodically` is that, once called, either could keep a process running indefinitely.

Unless I am misunderstanding how `every` and `periodically` function, it appears that they only keep running if another function is keeping the process alive. For example, my expectation of the code in [jedesmal](https://github.com/derwolfe/jedesmal), is that "I work" would be printed every second; until the application is terminated. In reality, it prints once, then exits.

In short - my expectations have not matched what the library does :-). Does this seem like a valuable feature to you? Or is there another way that you believe this should be dealt with outside of manifold?

Thank you!

**Comment by ztellman** [comment]
I've made the threads running `every` and `periodically` daemon threads because the much more typical complaint is "my process won't close" because of something a person did.  I understand why you'd expect the opposite behavior, but starting any non-daemon thread outside of your recurring tasks would accomplish what you're looking for.  I'd rather keep the "I want to stay alive" hint as a separate piece.

Any thoughts on this?

**Comment by derwolfe** [comment]
That seems completely reasonable. I've gone ahead and implemented this using a strategy exactly like you described. 

Thanks for your help! And, of course, thanks for manifold and aleph!

**Comment by vspinu** [comment]
I got  similar confusion.
 
> , but starting any non-daemon thread outside of your recurring tasks would accomplish what you're looking for.

What is the recommended way to  keep the `periodically` stream alive in an application? This is what I am using right now:

```cljs
(def ta (atom 0))
(def tp (s/periodically 100 #(swap! ta inc)))
(s/consume println tp)
;; then
(s/close! tp)
```

Thanks.

**Comment by derwolfe** [comment]
@vspinu I've implemented this using a demon thread.

For example:

```clojure
(defn staying-alive
  "Create a background thread to keep the entire process running."
  []
  (.start (Thread. (fn [] (.join (Thread/currentThread))) "I'm a background thread.")))

(defn ta (atom 0))
(defn tp (s/periodically 100 #(swap! ta inc)))

(defn run-world
  []
  (tp)
  (staying-alive))
```

**Comment by vspinu** [comment]
I see. Thanks.  Is there a particular benefit of this approach over the `s/consume` version?
